src/modules/menu/components/search/SearchBar.vue

```diff
@@ -1,117 +1,151 @@
+<script setup>
+import { computed, nextTick, onMounted, ref, watch } from 'vue'
+import { useStore } from 'vuex'
+
+import SearchResultList from '@/modules/menu/components/search/SearchResultList.vue'
+
+const store = useStore()
+
+const showResults = ref(false)
+const searchInput = ref(null)
+const searchValue = ref('')
+const results = ref(null)
+const selectedEntry = ref(null)
+
+const searchQuery = computed(() => store.state.search.query)
+const hasResults = computed(() => store.state.search.results.count())
+const isPhoneMode = computed(() => store.getters.isPhoneMode)
+
+watch(hasResults, (newValue) => {
+    // if an entry has been selected from the list, do not show the list again
+    // because the list has been hidden by onEntrySelected
+    if (!selectedEntry.value) {
+        showResults.value = newValue
+    }
+})
+
+watch(showResults, (newValue) => {
+    if (newValue && isPhoneMode.value && store.state.ui.showMenu) {
+        store.dispatch('toggleMenu')
+    }
+})
+
+watch(searchQuery, (newQuery) => {
+    searchValue.value = newQuery
+})
+
+onMounted(() => {
+    searchValue.value = searchQuery.value
+    searchInput.value.focus()
+})
+
+let debounceSearch = null
+const updateSearchQuery = (event) => {
+    selectedEntry.value = null
+    searchValue.value = event.target.value
+
+    clearTimeout(debounceSearch)
+    debounceSearch = setTimeout(() => {
+        store.dispatch('setSearchQuery', { query: event.target.value })
+    }, 50)
+}
+
+const onSearchInputFocus = () => {
+    if (hasResults.value) {
+        showResults.value = true
+    }
+}
+
+const clearSearchQuery = () => {
+    showResults.value = false
+    selectedEntry.value = null
+    searchValue.value = ''
+    store.dispatch('setSearchQuery', { query: '' })
+    searchInput.value.focus()
+}
+
+const closeSearchResults = () => {
+    showResults.value = false
+    searchInput.value.focus()
+}
+
+const goToFirstResult = () => {
+    if (hasResults.value) {
+        if (!showResults.value) {
+            showResults.value = true
+        }
+        // here we need to do the focus on nex tick if the result list was
+        // not displayed, otherwise the focus won't work (e.g result list
+        // is hidden and the user press the key down, it would then only show
+        // the list but not focus, to focus the user would need to press the key
+        // down again)
+        nextTick(() => {
+            results.value.$el.querySelector('[tabindex="0"]').focus()
```

Comment:
might be better to expose a function in each results' component instead of accessing and manipulating their inner HTML directly

Reply:
Done